python/meterpreter/meterpreter.py

```diff
@@ -680,10 +680,13 @@ def is_alive(self):
         return self.proc_h.poll() is None
 
     def read(self, length):
-        data = ''
+        data = bytes()
         stdout_reader = self.proc_h.stdout_reader
+        stderr_reader = self.proc_h.stderr_reader
         if stdout_reader.is_read_ready():
-            data = stdout_reader.read(length)
+            data += stdout_reader.read(length)
+        if stderr_reader.is_read_ready():
+            data += stderr_reader.read(length)
```

Comment:
Minor suggestion to read stderr before stdout like it is in the loop around L1320 and to still keep track of the length.

```suggestion
        if stderr_reader.is_read_ready() and length > 0:
            data += stderr_reader.read(length)
        if stdout_reader.is_read_ready() and (length - len(data)) > 0:
            data += stdout_reader.read(length - len(data))
```